geomstats/stratified_geometry/wald_space.py

```diff
@@ -0,0 +1,1024 @@
+        ----------
+        u : list of int, int
+            Either an integer or a set of labels.
+        v : list of int, int
+            Either an integer or a set of labels.
+
+        Returns
+        -------
+        is_separated : bool
+            A boolean determining whether u and v are separated by the split (i.e. if
+             they are not in the same part).
+        """
+        u = {u} if type(u) is int else set(u)
+        v = {v} if type(v) is int else set(v)
+        b1 = u.issubset(set(self.part1)) and v.issubset(set(self.part2))
+        b2 = v.issubset(set(self.part1)) and u.issubset(set(self.part2))
+        return b1 or b2
+
+    def get_part_towards(self, other):
+        """Return the part of this split that is directed toward the other split.
+
+        Each split contains part1 and part2, the parts that the corresponding edge in
+        the graph splits the set of labels into. Thus, one can think of the split as an
+        edge, where part1 points in the direction of the part of the tree where the
+        labels of part1 are contained, and part2 points in the other direction.
+        So, part1 points in the direction of ``other``, if it corresponds to an
+        edge that is contained in the tree that part1 points to, else part2 points in
+        the direction of ``split``.
+
+        Parameters
+        ----------
+        other : Split
+            The other split.
+
+        Returns
+        -------
+        part_towards : iterable
+            Return the part of the split ``self`` that points toward ``other_split``.
+        """
+        if other.part_contains(set(self.part1)):
+            return self.part2
+        return self.part1
+
+    def get_part_away_from(self, other):
+        """Return the part of this split that is directed away from other split.
+
+        Parameters
+        ----------
+        other : Split
+            The other split.
+
+        Returns
+        -------
+        part_that_does_not_point : iterable
+            Return the part of the split ``self`` that does not point toward
+            ``other``. See ``self.get_part_towards`` for further explanation.
+        """
+        if other.part_contains(set(self.part1)):
+            return self.part1
+        return self.part2
+
+    def is_compatible(self, other):
+        """Check whether this split is compatible with another split.
+
+        Two splits are compatible, if at least one intersection of the respective parts
+        of the splits is empty.
+
+        Parameters
+        ----------
+        other : Split
+            The other split.
+
+        Returns
+        -------
+        is_compatible_with : bool
+            Return ``True`` if the splits are compatible, else ``False``.
+        """
+        p1, p2 = set(self.part1), set(self.part2)
+        o1, o2 = set(other.part1), set(other.part2)
+        return sum([bool(s) for s in [p1 & o1, p1 & o2, p2 & o1, p2 & o2]]) < 4
+
+    def __eq__(self, other):
```

Comment:
In terms of class organization, I would probably move all the dunder methods (e.g. `__eq__`) to the beginning of the class, after `__init__`. Seems unnatural to find public methods first...